src/java/org/apache/cassandra/service/AbstractWriteResponseHandler.java

```diff
@@ -280,6 +289,8 @@ public void onFailure(InetAddressAndPort from, RequestFailureReason failureReaso
 
         failureReasonByEndpoint.put(from, failureReason);
 
+        logFailureOrTimeoutToIdealCLDelegate();
```

Comment:
Ouch looks like the failure path didn't decrement at all before?

I guess the only reason it gave some signal was because we would call expired explicitly if gossip thought the endpoint was down so it would be a non-zero metric.

Reply:
yes, I think so, for prolonged node loss cases detected by gossip we have another path via expired() method.